src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -592,13 +592,21 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
 
         internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, BindingDiagnosticBag diagnostics)
         {
-            // This check prevents redundant ManagedAddr diagnostics on the underlying pointer field of a fixed-size buffer
-            if (!IsFixedSizeBuffer)
+            var type = Type;
+            if (type.Kind != SymbolKind.PointerType && !IsFixedSizeBuffer)
             {
-                Type.CheckAllConstraints(DeclaringCompilation, conversions, ErrorLocation, diagnostics);
+                Type.CheckAllConstraints(DeclaringCompilation, new TypeConversions(this.ContainingAssembly.CorLibrary), ErrorLocation, DeclaringCompilation.AfterAccessorBindingDiagnostics);
             }
```

Comment:
You forgot to call the base `AfterAddingTypeMembersChecks`, which is causing your current test failures.

Reply:
Yes my bad. Thanks @333fred.